profiles/views.py

```diff
@@ -135,18 +135,34 @@ def get_queryset(self):
         return ViewedCompany.objects.filter(user=user_id).order_by("company_id")
 
 
-class CategoryList(ListAPIView):
-    model = Category
+class CategoryList(ListCreateAPIView):
     serializer_class = CategorySerializer
+    permission_classes = (UserIsAdmin,)
     queryset = Category.objects.all()
 
 
-class ActivityList(ListAPIView):
-    model = Activity
+class ActivityList(ListCreateAPIView):
     serializer_class = ActivitySerializer
+    permission_classes = (UserIsAdmin,)
     queryset = Activity.objects.all()
 
 
 class RegionListView(ListAPIView):
     serializer_class = RegionSerializer
     queryset = Region.choices
+
+
+class CategoryDetail(RetrieveUpdateDestroyAPIView):
+    serializer_class = CategorySerializer
+    permission_classes = (IsAdminUser,)
+
+    def get_queryset(self, pk=None):
+        return Category.objects.all()
```

Comment:
There's no need to override `get_queryset` method here. Just add `queryset`.
```suggestion
    queryset = Category.objects.all()
```

Reply:
Thank you for feedback. Done